### PR TITLE
[Config] Dependencies

### PR DESCRIPTION
**Pull request by runyasak**
# Overview
Because the dependencies are too old and it cannot run `start`. So, I decide to update them latest with configuration.

# Scope of work
* Update `react-scripts` to v4
* Update `react` and `react-dom` to latest
* Convert to `eslint`: `tslint` is deprecated
* Convert to `yarn`

Hope, this work. If you have any feedback or suggestion, do not hesitate to comment on this PR. 😎

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -1,10 +1,10 @@
-import * as React from "react";
-import AdministrativeDivision, { DivisionType } from "./AdministrativeDivision";
-import DatabaseTypeSelectList from "./DatabaseTypeSelectList";
-import SqlScriptExport from "./SqlScriptExport";
+import * as React from 'react'
```
Comment (aaronamm) [inline src/App.tsx]: Thanks for changing " to '.

File: src/AdministrativeDivision.tsx [hunk 2]
```diff
@@ -14,14 +13,39 @@ interface IProps {
     insertRecordSqlStatementPlaceHolder?: string;
 }
 
+const SqlScript: React.FunctionComponent<{ placeHolder: string }> = ({ placeHolder }) => {
```
Comment (aaronamm) [inline src/AdministrativeDivision.tsx:16]: Could you please explain the reason why you swap the order of these two components?

Comment (runyasak) [reply src/AdministrativeDivision.tsx:16]: @aaronamm 
I have followed the eslint. That we need to declare the variable before defining or using later. 😁

File: package.json [hunk 3]
```diff
@@ -5,18 +5,17 @@
...
+    "react-scripts": "4.0.0"
   },
   "scripts": {
     "start": "react-scripts start",
     "test": "./node_modules/.bin/jest",
-    "sass": "./node_modules/.bin/node-sass src/scss/style.scss public/style.css"
-  },
-  "eslintConfig": {
-    "extends": "react-app"
+    "sass": "./node_modules/.bin/node-sass src/scss/style.scss public/style.css",
```
Comment (aaronamm) [inline package.json]: I know this is the old code but you are working on this area. 
Could you simplify test and sass script command?
I think you can only use `test: jest` and `sass src/scss/style.scss public/style.css` and please also check if it work when you run these commands.
Thanks.

Comment (runyasak) [reply package.json]: I have simplified by implement with `react-scripts`.

**Comment by aaronamm** [comment]
@runyasak  Open for discussion.
I know ; (semicolon) is optional in JS/TS but I still like it. 
Could you please add a new commit for adding/update these files with their contents as following.

**.prettierrc.json, .editorconfig, .vscode/setting.json**

**.prettierrc.json**
```
{
    "semi": true
}
```


**.editorconfig**
```
# EditorConfig is awesome: https://EditorConfig.org

# top-most EditorConfig file
root = true

# Unix-style newlines with a newline ending every file
[*]
end_of_line = lf
insert_final_newline = true
charset = utf-8
indent_style = space
indent_size = 4
quote_type = double
trim_trailing_whitespace = true

# Matches multiple files with brace expansion notation
[*.{js, ts, css, scss}]
indent_size = 2
quote_type = single
```

**.vscode/setting.json** 
```
{
    "editor.defaultFormatter": "esbenp.prettier-vscode",
    "editor.formatOnSave": true
}
```

File: package.json (hunk 3 above)
Comment (aaronamm) [reply package.json]: Well done.

**Comment by runyasak** [comment]
@aaronamm 
Actually, I will contribute as you want and decision.

For my suggestion, we can define linter all rules just `eslint`.
Because `eslint` extension can now formatting on VSCode and preventing linter conflicts between `eslint` and `prettier`. 😁

**Pull request merged**